python/beeai_framework/adapters/openai/serve/server.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2025 © BeeAI a Series of LF Projects, LLC
+# SPDX-License-Identifier: Apache-2.0
+
+import contextlib
+from enum import StrEnum
+from typing import Any, Self
+
+import uvicorn
+from pydantic import BaseModel
+from typing_extensions import TypedDict, TypeVar, Unpack, override
+
+from beeai_framework.adapters.openai.serve.chat_completion.api import ChatCompletionAPI
+from beeai_framework.adapters.openai.serve.openai_model import OpenAIModel
+from beeai_framework.adapters.openai.serve.responses.api import ResponsesAPI
+from beeai_framework.agents.react import ReActAgent
+from beeai_framework.agents.requirement import RequirementAgent
+from beeai_framework.backend import ChatModel
+from beeai_framework.logger import Logger
+from beeai_framework.runnable import Runnable
+from beeai_framework.serve import MemoryManager
+from beeai_framework.serve.errors import FactoryAlreadyRegisteredError
+from beeai_framework.serve.server import Server
+from beeai_framework.utils import ModelLike
+from beeai_framework.utils.models import to_model
+
+logger = Logger(__name__)
+
+AnyRunnable = TypeVar("AnyRunnable", bound=Runnable[Any], default=Runnable[Any])
```

Comment:
I see such definitions in multiple places. Can you please create one directly in `beeai_framework/runnable.py` and use it here? Or maybe we could do that in another PR.